### PR TITLE
Gdr 2558.2

### PR DESCRIPTION
**Pull request by bczech**
# Description
## What changed?
Related JIRA issue: GDR-2558

## Why was it changed?
To allow for calculating sd

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

**Review by gladkia — APPROVED** [review]

File: tests/testthat/test-utils.R [hunk 1]
```diff
@@ -214,6 +214,13 @@ test_that("average_biological_replicates_dt works as expected", {
   expect_equal(dim(metrics_data), c(60, 29))
   expect_equal(dim(avg_metrics_data), c(40, 28))
   expect_true(!"Ligand" %in% names(avg_metrics_data))
+  
+  avg_metrics_data2 <- average_biological_replicates_dt(dt = metrics_data,
+                                                        var = "Ligand",
+                                                        prettified = TRUE,
+                                                        add_sd = TRUE)
+  
+  expect_equal(dim(avg_metrics_data2), c(40, 44))
```
Comment (j-smola) [inline tests/testthat/test-utils.R:223]: That is only one condition that has to be fulfill to be sure that `add_sd = TRUE` works as expected? 

Maybe also we should check whether new "sd" columns and "count" are added?

File: R/utils.R [hunk 2]
```diff
@@ -449,28 +448,47 @@ average_biological_replicates_dt <- function(
...
   geometric_average_fields <- intersect(geometric_average_fields, names(dt))
-  group_by <- setdiff(names(data),
-                      c(average_fields, var, id_cols))
-  group_by <- grep("Fit Type", group_by, invert = TRUE, value = TRUE)
+  group_by <- setdiff(names(data), c(average_fields, var, id_cols, "fit_type", "Fit Type"))
+  
+  if (add_sd) {
+    # Calculate standard deviation for both average_fields and geometric_average_fields
+    sd_fields <- paste0(average_fields, "_sd")
+    geo_sd_fields <- paste0(geometric_average_fields, "_sd")
```
[the remark this answers is not in the document]
Comment (bczech) [reply R/utils.R:473]: done

File: tests/testthat/test-utils.R (hunk 1 above)
Comment (bczech) [reply tests/testthat/test-utils.R:223]: done

File: NAMESPACE [hunk 3]
```diff
@@ -14,6 +14,7 @@ export(aggregate_assay)
 export(apply_bumpy_function)
 export(assert_choices)
 export(average_biological_replicates_dt)
+export(calc_sd)
```
Comment (j-smola) [inline NAMESPACE:17]: 👍🏻

File: R/utils.R [hunk 4]
```diff
@@ -733,3 +743,23 @@ get_additional_variables <- function(dt_list,
...
+#'
+#' @param x A numeric vector.
+#' @return The standard deviation of the vector if its length is greater than 1 or it is not numeric, otherwise 0.
+#' @examples
+#' calc_sd(c(1, 2, 3, 4, 5)) # Should return the standard deviation
+#' calc_sd(c(1)) # Should return 0
+#' calc_sd(numeric(0)) # Should return NA
+#' calc_sd(c("a", "b", "c")) # Should return NA
+#' @export
+calc_sd <- function(x) {
```
Comment (j-smola) [inline R/utils.R:758]: Please, add
```
#' @keywords package_utils
```

File: tests/testthat/test-utils.R [hunk 5]
```diff
@@ -557,4 +566,27 @@ test_that("capVals works as expected", {
   expect_equal(dt1c, dt2)
   
   expect_error(capVals(as.list(dt1)), "Must be a data.table")
-})
\ No newline at end of file
+})
+
+test_that("calc_sd returns correct standard deviation for vectors with length > 1", {
```
Comment (j-smola) [inline tests/testthat/test-utils.R:571]: Or maybe that coudl be one section - as all others function above?
```
test_that("cap calc_sd Vals works as expected", {
# calc_sd returns correct standard deviation for vectors with length > 1
  epect_equal(calc_sd(c(1, 2, 3, 4, 5)), sd(c(1, 2, 3, 4, 5), na.rm = TRUE))
  expect_equal(calc_sd(c(10, 20, 30)), sd(c(10, 20, 30), na.rm = TRUE))

# calc_sd returns 0 for vectors with length == 1 and numeric.
expect_equal(calc_sd(c(1)), 0)
...

})
```

**Review by j-smola — APPROVED** [review]

File: R/utils.R (hunk 4 above)
Comment (bczech) [reply R/utils.R:758]: done

File: tests/testthat/test-utils.R (hunk 5 above)
Comment (bczech) [reply tests/testthat/test-utils.R:571]: done

**Pull request merged**